catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java

```diff
@@ -215,8 +219,84 @@ private org.apache.hadoop.hive.metastore.api.Partition toHivePartition(
   }
 
   @Override
-  public boolean dropPartition(String partitionName) {
-    throw new UnsupportedOperationException();
+  public boolean dropPartition(String partitionName, boolean ifExists)
+      throws NoSuchPartitionException {
+    try {
+      // check the partition exists
+      if (!ifExists) {
+        table
+            .clientPool()
+            .run(c -> c.getPartition(table.schemaName(), table.name(), partitionName));
+      }
+
+      // get all partitions that will be deleted
+      Table hiveTable = table.clientPool().run(c -> c.getTable(table.schemaName(), table.name()));
+      List<org.apache.hadoop.hive.metastore.api.Partition> partitions =
+          table
+              .clientPool()
+              .run(
+                  c ->
+                      c.listPartitions(
+                          table.schemaName(),
+                          table.name(),
+                          getFilterPartitionList(hiveTable, partitionName),
+                          (short) -1));
+
+      // delete partitions iteratively
+      for (org.apache.hadoop.hive.metastore.api.Partition partition : partitions) {
+        table
+            .clientPool()
+            .run(
+                c ->
+                    c.dropPartition(
+                        partition.getDbName(),
+                        partition.getTableName(),
+                        partition.getValues(),
+                        false));
+      }
+    } catch (UnknownTableException e) {
+      if (!ifExists) {
+        throw new NoSuchTableException(
+            e, "Hive table %s does not exist in Hive Metastore", table.name());
+      }
+
+    } catch (NoSuchObjectException e) {
+      if (!ifExists) {
+        throw new NoSuchPartitionException(
+            e, "Hive partition %s does not exist in Hive Metastore", partitionName);
+      }
+
+    } catch (TException | InterruptedException e) {
+      throw new RuntimeException(
+          "Failed to get partition "
+              + partitionName
+              + " of table "
+              + table.name()
+              + "from Hive Metastore",
+          e);
+    }
+    return true;
+  }
+
+  @Override
+  public boolean dropPartitions(List<String> partitionNames, boolean ifExists)
+      throws NoSuchPartitionException, UnsupportedOperationException {
+    if (partitionNames.size() > 1) {
+      throw new UnsupportedOperationException("Only one partition is supported");
+    }
+    return dropPartition(partitionNames.get(0), ifExists);
+  }
+
+  private List<String> getFilterPartitionList(Table dropTable, String partitionSpec) {
+    Map<String, String> partMap = new HashMap<>();
+    String[] parts = partitionSpec.split("/");
+    for (String part : parts) {
+      String[] keyValue = part.split("=");
+      if (keyValue.length == 2) {
+        partMap.put(keyValue[0], keyValue[1]);
+      }
```

Comment:
If keyValue.length == 2 not matched, at least some message need be written to the log

Reply:
Thanks! It's a good advice.
If keyValue.length == 2 not matched, that means the user didn't deliver a correct partition name to Gravitino (partition should be example like "log_date=2024"), so I'll throw a NoSuchPartitionException in this place

Reply:
done